src/test/scala/chisel3/TypeEquivalenceSpec.scala

```diff
@@ -224,45 +241,94 @@ class TypeEquivalenceSpec extends AnyFlatSpec {
     // TODO should there be some form of structural typing for ChiselEnums?
-    x.findFirstTypeMismatch(z, false, true) should be(Some(": Left (Fizz) and Right (Buzz) have different types."))
+    x.findFirstTypeMismatch(z, false, true, true) should be(
+      Some(": Left (Fizz) and Right (Buzz) have different types.")
+    )
   }
 
   it should "support comparing Analogs" in {
     val x = Analog(8.W)
     val y = Analog(8.W)
     val z = Analog(4.W)
-    x.findFirstTypeMismatch(y, true, true) should be(None)
-    x.findFirstTypeMismatch(z, true, true) should be(
+    x.findFirstTypeMismatch(y, true, true, true) should be(None)
+    x.findFirstTypeMismatch(z, true, true, true) should be(
       Some(": Left (Analog<8>) and Right (Analog<4>) have different widths.")
     )
-    x.findFirstTypeMismatch(z, true, false) should be(None)
+    x.findFirstTypeMismatch(z, true, false, true) should be(None)
   }
 
   it should "support DontCare" in {
-    DontCare.findFirstTypeMismatch(DontCare, true, true) should be(None)
+    DontCare.findFirstTypeMismatch(DontCare, true, true, true) should be(None)
   }
 
   it should "support AsyncReset" in {
-    AsyncReset().findFirstTypeMismatch(AsyncReset(), true, true) should be(None)
-    AsyncReset().findFirstTypeMismatch(Bool(), true, true) should be(
+    AsyncReset().findFirstTypeMismatch(AsyncReset(), true, true, true) should be(None)
+    AsyncReset().findFirstTypeMismatch(Bool(), true, true, true) should be(
       Some(": Left (AsyncReset) and Right (Bool) have different types.")
     )
-    AsyncReset().findFirstTypeMismatch(Clock(), true, true) should be(
+    AsyncReset().findFirstTypeMismatch(Clock(), true, true, true) should be(
       Some(": Left (AsyncReset) and Right (Clock) have different types.")
     )
-    AsyncReset().findFirstTypeMismatch(Reset(), true, true) should be(
+    AsyncReset().findFirstTypeMismatch(Reset(), true, true, true) should be(
       Some(": Left (AsyncReset) and Right (Reset) have different types.")
     )
   }
 
   it should "support Clock" in {
-    Clock().findFirstTypeMismatch(Clock(), true, true) should be(None)
+    Clock().findFirstTypeMismatch(Clock(), true, true, true) should be(None)
   }
 
   it should "support abstract Reset" in {
-    Reset().findFirstTypeMismatch(Reset(), true, true) should be(None)
+    Reset().findFirstTypeMismatch(Reset(), true, true, true) should be(None)
+  }
+
+  it should "support Probe" in {
+    Probe(Bool()).findFirstTypeMismatch(Probe(Bool()), true, true, true) should be(None)
+  }
+
+  it should "support RWProbe" in {
+    RWProbe(Bool()).findFirstTypeMismatch(RWProbe(Bool()), true, true, true) should be(None)
+  }
+
+  it should "detect differences between Probe and Not-Probe" in {
+    Probe(Bool()).findFirstTypeMismatch(Bool(), true, true, true) should be(
+      Some(
+        ": Left (Bool with probeInfo: Some(writeable=false, color=None)) and Right (Bool with probeInfo: (None) have different probeInfo."
```

Comment:
```suggestion
        ": Left (Bool with probeInfo: Some(writeable=false, color=None)) and Right (Bool with probeInfo: (None)) have different probeInfo."
```
This seems to be missing a parentheses.